## Intro
This is a eda on aggregated version of CMU covid-symptom survey data. 

The original data had 10M records but htis aggregated version has 673k aggregated records

In [ ]:
import pandas as pd
import numpy as np
import os

In [2]:
# set working directories
data_root = "/home/ubuntu/datasets/covid-symptom-challenge/"
code_root = "/home/ubuntu/codebase/my_github/covid-symptom-challenge/"

In [9]:
pd.set_option('display.max_columns',None)

In [3]:
# decompress the data
os.chdir(data_root)
#!gzip -d overall-county-smoothed.csv.gz

In [4]:
!ls -l --block-size=M

total 446M
-rw-r--r-- 1 ubuntu ubuntu 446M Sep 12 17:00 overall-county-smoothed.csv


In [5]:
os.chdir(code_root)

In [10]:
df = pd.read_csv(f'{data_root}/overall-county-smoothed.csv', parse_dates=True)
df.head()

date state_code  fips  gender age_bucket  summed_n  smoothed_pct_cli  \
0  2020-08-25         ak  2020  female    overall       197            0.0000   
1  2020-08-24         ak  2020  female    overall       210            0.0000   
2  2020-08-23         ak  2020  female    overall       220            0.4675   
3  2020-08-22         ak  2020  female    overall       233            0.4414   
4  2020-08-21         ak  2020  female    overall       221            0.4654   

   smoothed_pct_ili  smoothed_pct_cli_anosmia_ageusia  smoothed_pct_hh_cli  \
0            0.0000                            1.5366               0.0000   
1            0.0000                            1.4414               0.0000   
2            0.4675                            1.8435               0.4545   
3            0.4414                            1.7406               0.8584   
4            0.4654                            1.8351               1.3575   

   smoothed_pct_cmnty_cli  smoothed_pct_hh_fever  smoothed_pct_hh_sore_throat  \
0                 16.4529                 0.0000                      11.1675   
1                 15.4954                 0.0000                      11.4286   
2                 17.0541                 0.4545                      10.0000   
3                 18.8978                 0.8584                       7.7253   
4                 18.5664                 1.3575                       6.3348   

   smoothed_pct_hh_cough  smoothed_pct_hh_shortness_of_breath  \
0                12.1827                               3.0457   
1                15.7143                               3.3333   
2                16.8182                               4.5455   
3                15.4506                               4.2918   
4                14.9321                               4.9774   

   smoothed_pct_hh_difficulty_breathing  smoothed_mean_hh_cli_ct  \
0                                2.5381                   0.1364   
1                                2.8571                   0.1032   
2                                4.0909                   0.1658   
3                                3.8627                   0.1524   
4                                4.5249                   0.1381   

   smoothed_mean_cmnty_cli_ct  smoothed_pct_self_fever  \
0                      8.7567                   0.0000   
1                      8.2255                   0.0000   
2                      7.9530                   0.4675   
3                      7.5615                   0.4414   
4                      7.9495                   0.4654   

   smoothed_pct_self_cough  smoothed_pct_self_shortness_of_breath  \
0                   9.7249                                 2.0500   
1                  12.1439                                 2.4311   
2                  13.9295                                 3.7232   
3                  12.7892                                 3.5375   
4                  12.5787                                 3.7295   

   smoothed_pct_self_difficulty_breathing  \
0                                  1.0348   
1                                  1.4787   
2                                  2.8141   
3                                  2.6791   
4                                  3.2770   

   smoothed_pct_self_tiredness_or_exhaustion  \
0                                    14.4047   
1                                    13.5603   
2                                    12.5153   
3                                    13.6878   
4                                    13.9786   

   smoothed_pct_self_nasal_congestion  smoothed_pct_self_runny_nose  \
0                             12.3020                       10.7987   
1                             11.5674                       10.1639   
2                             10.6390                       11.5850   
3                             11.5311                       11.9291   
4                             12.1572                       11.6718   

   smoothed_pct

In [7]:
df.shape

(673340, 104)

The csv file contains 673k rows and 104 columns

## Dimensional columns


### date 
- the date the data was collected (in US, Pacific time)
- collection over 4.5 months, no missing date

In [23]:
df.date.max(), df.date.min()

('2020-08-25', '2020-04-12')

In [47]:
df['date']= pd.to_datetime(df['date']) 

In [48]:
df_dates = df[['date','state_code']].groupby('date', as_index=False).count()
df_dates.head()

date  state_code
0 2020-04-12         859
1 2020-04-13        2752
2 2020-04-14        5177
3 2020-04-15        5272
4 2020-04-16        5416

In [59]:
# confirms no missing dates in the data
sum(df_dates.date.diff()/ np.timedelta64(1, 'D') > 1)

0

### fips 
- the FIPS county ID

In [62]:
len(df.fips.unique()), df.fips.tail()

(2173,
 673335    56041
 673336    56041
 673337    56041
 673338    56041
 673339    56041
 Name: fips, dtype: int64)

### state_code 
- the 2 letter state code
- covers all 51 states, no missing dates for all 51 states

In [64]:
len(df.state_code.unique())

51

In [67]:
df[['state_code','date']].groupby('state_code').nunique()

state_code  date
state_code                  
ak                   1   136
al                   1   136
ar                   1   136
az                   1   136
ca                   1   136
co                   1   136
ct                   1   136
dc                   1   136
de                   1   136
fl                   1   136
ga                   1   136
hi                   1   136
ia                   1   136
id                   1   136
il                   1   136
in                   1   136
ks                   1   136
ky                   1   136
la                   1   136
ma                   1   136
md                   1   136
me                   1   136
mi                   1   136
mn                   1   136
mo                   1   136
ms                   1   136
mt                   1   136
nc                   1   136
nd                   1   135
ne                   1   136
nh                   1   136
nj                   1   136
nm                   1   136
nv                   1   136
ny                   1   136
oh                   1   136
ok                   1   136
or                   1   136
pa                   1   136
ri                   1   136
sc                   1   136
sd                   1   135
tn                   1   136
tx                   1   136
ut                   1   136
va                   1   136
vt                   1   136
wa                   1   136
wi                   1   136
wv                   1   136
wy                   1   135

### gender
gender of survey respondent. Possible values:
- male
- female
- non-binary or other. This category unfortunately must be excluded from the data; because it is uncommonly selected, releasing aggregates for it could make respondents identifiable. Even if the category was not reported separately, its values can be deduced by comparing the other categories, making it possible to calculate the responses submitted by individual respondents who selected this option. To avoid risk of re-identification and harm, we remove these responses from this aggregate data.
- overall (aggregation of all genders)

findings - skewed distribution, most of the data lost in the aggregation

In [69]:
df[['gender','fips']].groupby('gender').count()

fips
gender         
female   144190
male      85441
overall  443709

### age_bucket 
age_bucket of survey respondent. 
Possible values:
- 18-34
- 35-54
- 55+
- overall (aggregation of all ages)

findings - 2/3rd of the data lost in aggregation

In [70]:
df[['age_bucket','fips']].groupby('age_bucket').count()

fips
age_bucket        
18-34        70333
35-54        95332
55+          97537
overall     410138

### n
- sample size of the date, fips, gender, and age_bucket aggregation.
Note: Rows with n < 50 are not reported in this dataset.
unavailable, may not be part of smoothed datasets

In [74]:
# unavailable, may not be part of smoothed datasets
# df[['n']]

### summed_n 
in smoothed datasets, summed_n is the sum of n over the days used to calculate the smoothed estimates (7 day lagging sum)

unsure what it means yet, will have to revisit

In [75]:
df.summed_n

0         197
1         210
2         220
3         233
4         221
         ... 
673335     66
673336     58
673337     71
673338     93
673339     97
Name: summed_n, Length: 673340, dtype: int64

### weight_sums
Sum of the weights associated with the calculations

unavailable, might be so due to aggregated data

In [77]:
# unavailable in aggregated data
# df.weight_sums

## Signals

### pct_cli 
- Estimated percentage of people with COVID-like illness, which is defined as exhibiting a fever along with cough or shortness of breath or difficulty breathing.

### pct_ili 
- Estimated percentage of people with Influenza-like illness, which is defined as exhibiting a fever with sore throat or cough pct_cli_anosmia_ageusia - estimated percentage of people with (1) COVID-like illness, which is defined as exhibiting a fever along with cough or shortness of breath or difficulty breathing OR (2) anosmia OR (3) ageusia

### pct_hh_cli 
- Estimated percentage of people reporting COVID-like illness in their household

### pct_cmnty_cli
- Estimated percentage of people reporting knowing someone with COVID-like illness in their local community. This data only exists from 2020-04-16 onwards.

###  pct_hh_fever
- Estimated percentage of people reporting that they or someone in their household experienced a fever in the past 24 hours

### pct_hh_sore_throat
- Estimated percentage of people reporting that they or someone in their household experienced a sore throat in the past 24 hours

###  pct_hh_cough 
- Estimated percentage of people reporting that they or someone in their household experienced a cough in the past 24 hours

### pct_hh_shortness_of_breath 
- Estimated percentage of people reporting that they or someone in their household experienced shortness of breath in the past 24 hours

### pct_hh_difficulty_breathing 
- Estimated percentage of people reporting that they or someone in their household experienced difficulty breathing in the past 24 hours

### mean_hh_cli_ct 
- Estimated average number of people in respondent’s household who are experiencing a fever and at least one of the following: a sore throat, a cough, shortness of breath, or difficulty breathing. This data only exists from 2020-04-16 onwards.

### mean_cmnty_cli_ct 
- Estimated average number of people in respondent’s local community who are experiencing a fever and at least one of the following: a sore throat, a cough, shortness of breath, or difficulty breathing. This data only exists from 2020-04-16 onwards.

###  pct_self_fever 
- Estimated percentage of people reporting that they have experienced a fever in the past 24 hours pct_self_cough - Estimated percentage of people reporting that they have experienced a cough in the past 24 hours

### pct_self_shortness_of_breath 
- Estimated percentage of people reporting that they have experienced shortness of breath in the past 24 hours 

### pct_self_difficulty_breathing 
- Estimated percentage of people reporting that they have experienced difficulty breathing in the past 24 hours

### pct_self_tiredness_or_exhaustion 
- Estimated percentage of people reporting that they have experienced tiredness or exhaustion in the past 24 hours

### pct_self_nasal_congestion 
- Estimated percentage of people reporting that they have experienced nasal congestion in the past 24 hours

### pct_self_runny_nose 
- Estimated percentage of people reporting that they have experienced a runny nose in the past 24 hours

### pct_self_muscle_joint_aches 
- Estimated percentage of people reporting that they have experienced muscle or joint aches in the past 24 hours

### pct_self_sore_throat 
- Estimated percentage of people reporting that they have experienced a sore throat in the past 24 hours

### pct_self_persistent_pain_pressure_in_chest 
- Estimated percentage of people reporting that they have experienced persistent pain or pressure in heir chest in the past 24 hours

### pct_self_nausea_vomiting 
- Estimated percentage of people reporting that hey have experienced nausea or vomiting in the past 24 hours

### pct_self_diarrhea 
- Estimated percentage of people reporting that they have experienced diarrhea in the past 24 hours

### pct_self_anosmia_ageusia 
- Estimated percentage of people reporting that they have experienced loss of smell or taste in the past 24 hours

### pct_self_other 
- Estimated percentage of people reporting that they have experienced another, not aforementioned symptom in the past 24 hours

### pct_self_none_of_above 
- Estimated percentage of people reporting that they have have not experienced any of the above symptoms in the past 24 hours

### pct_self_multiple_symptoms 
- Estimated percentage of people reporting that they have experienced multiple of the above symptoms in the past 24 hours

### pct_tested_and_positive 
- Estimated percentage of people reporting that they have tested positive for COVID-19. Warning: This survey item (B5) is only presented to respondents who report they are currently experiencing symptoms, so most responses are blank. (Wave 4 of the survey, not yet included in this data, will ask every respondent if they have been tested.)

### pct_tested_and_negative 
- Estimated percentage of people reporting that they have tested negative for COVID-19

### pct_tested_no_result 
- Estimated percentage of people reporting that they have tested for COVID-19 but have not received a result

### pct_could_not_get_tested 
- Estimated percentage of people reporting that they tried to get tested but could not get a test

### pct_did_not_try_to_get_tested 
- Estimated percentage of people reporting that they have not tried to get tested for COVID-19

### pct_worked_outside_home 
- Estimated percentage of people reporting that they have gone to work outside of their home

### pct_avoid_contact_all_or_most_time 
- Estimated percentage of people reporting that they are intentionally avoiding contact with people “All of the time” or “most of the time”

### mean_outside_hh_contact_at_work_ct 
- Estimated average number of people the respondent has had direct contact with outside of their household at work in the past 24 hours.

### mean_outside_hh_contact_shopping_ct 
- Estimated average number of people the respondent has had direct contact with outside of their household while shopping for groceries and other essentials in the past 24 hours.

### mean_outside_hh_contact_in_social_gatherings_ct 
- Estimated average number of people the respondent has had direct contact with outside of their household in social gatherings in the past 24 hours. 

### pct_contact_covid_positive 
- Estimated percentage of people reporting that they had direct contact with anyone who recently tested positive for COVID-19.

### pct_diabetes 
- Estimated percentage of people reporting that they have been diagnosed with diabetes

### pct_cancer 
- Estimated percentage of people reporting that they have been diagnosed with cancer (other than skin cancer)

### pct_heart_disease 
- Estimated percentage of people reporting that they have been diagnosed with a heart disease

### pct_high_blood_pressure 
- Estimated percentage of people reporting that they have been diagnosed with high blood pressure

### pct_asthma 
- Estimated percentage of people reporting that they have been diagnosed with asthma

### pct_chronic_lung_disease 
- Estimated percentage of people reporting that they have been diagnosed with a chronic lung disease such as COPD or emphysema

### pct_kidney_disease 
- Estimated percentage of people reporting that they have been diagnosed with kidney disease

### pct_autoimmune_disorder 
- Estimated percentage of people reporting that hey have been diagnosed with an autoimmune disorder such as rheumatoid arthritis or Crohn’s disease

### pct_no_above_medical_conditions 
- Estimated percentage of people reporting hat they have not been diagnosed with any of the above conditions

### pct_multiple_medical_conditions 
- Estimated percentage of people reporting hat they have been diagnosed with multiple of the above conditions


In [13]:
# remove the first 6 dimentional columns
df.columns[6:].str.contains('smoothed')

array([ True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True])

In [14]:
df.columns[6:].str.contains('weighted')

array([False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True])

In [18]:
df_weighted = df.iloc[:,df.columns.str.contains('weighted')]
df_weighted.columns

Index(['smoothed_pct_cli_weighted', 'smoothed_pct_ili_weighted',
       'smoothed_pct_cli_anosmia_ageusia_weighted',
       'smoothed_pct_hh_cli_weighted', 'smoothed_pct_cmnty_cli_weighted',
       'smoothed_pct_hh_fever_weighted',
       'smoothed_pct_hh_sore_throat_weighted',
       'smoothed_pct_hh_cough_weighted',
       'smoothed_pct_hh_shortness_of_breath_weighted',
       'smoothed_pct_hh_difficulty_breathing_weighted',
       'smoothed_mean_hh_cli_ct_weighted',
       'smoothed_mean_cmnty_cli_ct_weighted',
       'smoothed_pct_self_fever_weighted', 'smoothed_pct_self_cough_weighted',
       'smoothed_pct_self_shortness_of_breath_weighted',
       'smoothed_pct_self_difficulty_breathing_weighted',
       'smoothed_pct_self_tiredness_or_exhaustion_weighted',
       'smoothed_pct_self_nasal_congestion_weighted',
       'smoothed_pct_self_runny_nose_weighted',
       'smoothed_pct_self_muscle_joint_aches_weighted',
       'smoothed_pct_self_sore_throat_weighted',
       'smoothed_p